application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -850,7 +851,7 @@ define('diagram-image-editor', ['xwiki-utils', 'diagram-link-handler', 'draw.io'
             // when saving the diagram (we want to save the attachment reference not the attachment URL).
             let resourceReference = {
               type: 'attach',
-              reference: XWiki.Model.serialize(new XWiki.AttachmentReference(filename))
+              reference: XWiki.Model.serialize(new XWiki.AttachmentReference(filename, xm.documentReference))
```

Comment:
Doesn't this lead to the absolute attachment reference being saved in the diagram content? We should be saving relative references whenever possible, as this makes the content portable: you should be able to export the diagram page and import it in a different wiki or subwiki and it should still work, provided you have the Diagram application installed. So I would try to fix this when the diagram is displayed, by resolving the relative attachment reference relative to the diagram document, or, if that's not possible or complex, at least don't include the wiki in the saved attachment reference.